keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,973 @@
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+User deployed a service in a cluster and to ensure its high availability, 
+User created PDB for this deployment and set `.spec.minAvailable` to 3. User wanted the PDB to be 
+guaranteed even in case of scheduler preemption.
+
+#### Story 2
+User created a Tensorflow distributed job that requires a minimum of 5 workers running.
+User created PDB for the job and set `.spec.minAvailable` to 5. User wanted the PDB to be 
+guaranteed even in case of scheduler preemption to ensure the stability of the whole job.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+If a user sets a PodDisruptionBudget (PDB) for some low-priority pods and sets the `AllowDisruptionByPriorityGreaterThanOrEqual` 
+to `PriorityClass`, high-priority (less than the value of allowDisruptionByPriorityGreaterThanOrEqual) pods will not be able 
+to violate the PDB and preempt these pods during the scheduling process. This may result in high-priority (less than the value of allowDisruptionByPriorityGreaterThanOrEqual) pods being unable to schedule while low-priority pods continue to run normally. 
+
+Although the above situations may arise, due to the fact that PriorityClass is created and managed by 
+cluster administrators with no permission for application owners to perform actions, administrators 
+are able to uniformly configure according to the requirements. Additionally, implementation will 
+include the addition of additional logging or event descriptions to clearly inform the user of the 
+reason why preemption did not occur.
```

Comment:
Also system-critical will always be able to preempt.

Reply:
Added.

Reply:
Good to explicitly document, but that should not require special coding, since that is already the highest priority ?